internal/audit/cluster_checks.go

```diff
@@ -214,3 +218,58 @@ func makeCheckClusterHighHAAssets(haAssetsThreshold int) checkFunc {
 		return Pass, nil
 	}
 }
+
+func checkClusterNamesForWhitespace(reader *archive.Reader, examples *ExamplesCollection) (Outcome, error) {
+
+	for _, clusterName := range reader.GetClusterNames() {
+		if strings.ContainsFunc(clusterName, unicode.IsSpace) {
+			examples.add("Cluster name '%s' contains whitespace", clusterName)
+		}
+	}
+
+	if examples.Count() > 0 {
+		logCritical("Found %d clusters with names containing spaces", examples.Count())
+		return Fail, nil
+	}
+
+	return Pass, nil
+}
+
+func checkLeafnodeServerNamesForWhitespace(r *archive.Reader, examples *ExamplesCollection) (Outcome, error) {
+
+	for _, clusterName := range r.GetClusterNames() {
+		clusterTag := archive.TagCluster(clusterName)
+
+		invalidLeafnodes := map[string]struct{}{}
```

Comment:
Rename to `leafNodesWithWhitespace` or maybe `leafNodeNames`?
"Invalid" is a bit misleading.